internal/services/compute/shared_image_data_source.go

```diff
@@ -174,6 +205,10 @@ func dataSourceSharedImageRead(d *pluginsdk.ResourceData, meta interface{}) erro
 			if err := d.Set("purchase_plan", flattenGalleryImageDataSourcePurchasePlan(props.PurchasePlan)); err != nil {
 				return fmt.Errorf("setting `purchase_plan`: %+v", err)
 			}
+
+			if err := flattenAndSetGalleryImageFeatures(d, model.Properties.Features); err != nil {
```

Comment:
We typically only define a dedicated flatten function for blocks or if multiple resources have a use for it. Since these are all top level properties could you please in-line the contents of the flatten function in here?

Reply:
👍